snapcraft/parts/plugins/matter_sdk_plugin.py

```diff
@@ -111,47 +111,48 @@ def get_build_snaps(self) -> Set[str]:
 
     @overrides
     def get_build_commands(self) -> List[str]:
-        options = cast(MatterPluginProperties, self._options)
+        options = cast(MatterSdkPluginProperties, self._options)
         commands = []
 
         if self.snap_arch == "arm64":
             commands.extend(
                 [
-                    f"wget --no-verbose https://github.com/project-chip/zap/releases/download/"
-                    f"{options.matter_zap_version}/zap-linux-{self.snap_arch}.zip",
+                    "wget --no-verbose https://github.com/project-chip/zap/releases/download/"
+                    f"{options.matter_sdk_zap_version}/zap-linux-{self.snap_arch}.zip",
                     f"unzip -o zap-linux-{self.snap_arch}.zip",
                     "echo 'export ZAP_INSTALL_PATH=$PWD'",
                 ]
             )
 
-        # Clone Matter repository if not present
+        # Clone Matter SDK repository if not present
         commands.extend(
             [
-                "if [ ! -d matter ]; then",
-                f"    git clone --depth 1 -b {options.matter_sdk_version} {MATTER_REPO} matter;",
-                "else",
-                "    echo 'Matter repository already exists, skip clone';",
-                "fi",
-                "cd matter;",
-            ]
+            "if [ ! -d matter ]; then",
+            "    git init matter-sdk",
+            "    cd matter-sdk",
+            f"   git remote add origin {MATTER_SDK_REPO}",
+            f"   git fetch --depth 1 origin {options.matter_sdk_version}",
+            "    git checkout FETCH_HEAD",
+            "else",
+            "    echo 'Matter SDK repository already exists, skip clone'",
+            "    cd matter-sdk;",
+            "fi",]
```

Comment:
Why is the repo cloned inside a subdirectory?

Reply:
It was intended to organize files, but the matter-sdk directory can be removed, as all files can stay under `root/part/matter-sdk/build/`.